### PR TITLE
Bump sra toolkit version

### PR DESCRIPTION
**Pull request by mvdbeek**
There are occasional connection issues when downloading datasets,
so this an easy first try.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/sra-tools/fasterq_dump.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<tool id="fasterq_dump" name="Faster Download and Extract Reads in FASTQ" version="@VERSION@.4" profile="16.01">
+<tool id="fasterq_dump" name="Faster Download and Extract Reads in FASTQ" version="@VERSION@.4+galaxy1" profile="16.01">
```
Comment (wm75) [inline tools/sra-tools/fasterq_dump.xml]: I guess the `.4` should be dropped here, and in `fastq_dump`and `sam_dump`?

Comment (mvdbeek) [reply tools/sra-tools/fasterq_dump.xml]: No, then the tool wouldn't be newer in the tool panel.

Comment (wm75) [reply tools/sra-tools/fasterq_dump.xml]: Damn! But it won't be either way since 2.10.3.4 < 2.10.4 ?

Comment (mvdbeek) [reply tools/sra-tools/fasterq_dump.xml]: Damn, you're right.

Comment (mvdbeek) [reply tools/sra-tools/fasterq_dump.xml]: I guess we'll need to keep version at 2.10.4 and add +galaxy1

Comment (wm75) [reply tools/sra-tools/fasterq_dump.xml]: Yeah, this is really unfortunate, but I can't think of another way for now either. Lets hope that the package will arrive at 2.10.4 soon :)

**Comment by mvdbeek** [comment]
This is now stuck on getting sra-tools bioconda recipe updated (https://github.com/bioconda/bioconda-recipes/pull/20503)

File: tools/sra-tools/sra_macros.xml [hunk 2]
```diff
@@ -1,8 +1,11 @@
 <macros>
-    <token name="@VERSION@">2.10</token>
+    <token name="@VERSION@">2.10.4+galaxy1</token>
```
Comment (nsoranzo) [inline tools/sra-tools/sra_macros.xml]:
```suggestion
    <token name="@VERSION@">2.10.4</token>
```

File: tools/sra-tools/sam_dump.xml [hunk 3]
```diff
@@ -1,9 +1,11 @@
-<tool id="sam_dump" name="Download and Extract Reads in BAM" version="@VERSION@.4">
+<tool id="sam_dump" name="Download and Extract Reads in BAM" version="@VERSION@">
```
Comment (nsoranzo) [inline tools/sra-tools/sam_dump.xml]:
```suggestion
<tool id="sam_dump" name="Download and Extract Reads in BAM" version="@VERSION@+galaxy1">
```

**Pull request merged**